src/dialogs/RemoteDebugDialog.cpp

```
@@ -155,6 +163,76 @@ void RemoteDebugDialog::setDebugger(QString debugger)
 }
 
+void RemoteDebugDialog::on_actionRemove_item_triggered()
+{
+    QListWidgetItem *item = ui->recentsIpListWidget->currentItem();
+
+    if (item == nullptr)
+        return;
+
+    QVariant data = item->data(Qt::UserRole);
+    QString sitem = data.toString();
+
+    QSettings settings;
+    QStringList ips = settings.value("recentIpList").toStringList();
+    ips.removeAll(sitem);
+    settings.setValue("recentIpList", ips);
+
+    ui->recentsIpListWidget->takeItem(ui->recentsIpListWidget->currentRow());
+
+    ui->ipEdit->clear();
+
+}
+
+void RemoteDebugDialog::on_actionClear_all_triggered()
+{
+    QSettings settings;
+    QStringList ips = settings.value("recentIpList").toStringList();
+    ips.clear();
+
+    ui->recentsIpListWidget->clear();
+    settings.setValue("recentIpList", ips);
+    ui->ipEdit->clear();
+}
+
+bool RemoteDebugDialog::fillRecentIpList()
+{
+    QSettings settings;
+
+    QStringList ips = settings.value("recentIpList").toStringList();
+    QMutableListIterator<QString> it(ips);
+    while (it.hasNext()) {
+        const QString ip = it.next();
+        const QString text = QString("%1").arg(ip);
+        QListWidgetItem *item = new QListWidgetItem(
+            text
+        );
+        item->setData(Qt::UserRole, ip);
+        ui->recentsIpListWidget->addItem(item);
+    }
+
+    settings.setValue("recentIpList", ips);
+    return !ips.isEmpty();
+}
+
+void RemoteDebugDialog::on_recentsIpListWidget_itemClicked(QListWidgetItem *item)
+{
+    QVariant data = item->data(Qt::UserRole);
+    QString ipport = data.toString();
+    if (!ipport.indexOf(GDB_URI_PREFIX)) {
+        ui->debuggerCombo->setCurrentIndex(ui->debuggerCombo->findText(GDBSERVER));
+        int last_colon = ipport.lastIndexOf(QString(":"));
+        QString port_temp = ipport.mid(last_colon + 1, ipport.length());
+        QString ip_temp = ipport.mid(6, ipport.length() - port_temp.length() - 7);
```

why "_temp"?

---

Try not to use hardcoded (6, -7)

---

"_temp" because I was struggling to name my variables good :P 

I was looking for a workaround of not using hardcoded values but I failed to find any. I'll give it another try :)